pkg/quota/evaluator/core/pods.go

```diff
@@ -269,12 +284,45 @@ func isTerminating(pod *api.Pod) bool {
 }
 
 // QuotaPod returns true if the pod is eligible to track against a quota
-func QuotaPod(pod *api.Pod) bool {
-	return !(api.PodFailed == pod.Status.Phase || api.PodSucceeded == pod.Status.Phase)
+// A pod is eligible for quota, unless any of the following are true:
+//  - pod has a terminal phase (failed or succeeded)
+//  - pod has been marked for deletion and grace period has expired.
+func QuotaPod(pod *api.Pod, clock clock.Clock) bool {
+	// if pod is terminal, ignore it for quota
+	if api.PodFailed == pod.Status.Phase || api.PodSucceeded == pod.Status.Phase {
+		return false
+	}
+	// if pods are stuck terminating (for example, a node is lost), we do not want
```

Comment:
This may be obvious to you, but not to me.  deletiontimestamp non-nil means its terminating?  I thought an update had to come back to recognize it as a terminating.

If you mean, "deleted pods that should be gone should not be charged", then this code makes sense to me.

Reply:
it means "deleted pods that should be gone should not be charged", i will look to clarify comment.